hybrid/compartment_examples/inter_comp_call/secure-redirect_clr/compartments-redirect_clr.S

```diff
@@ -0,0 +1,51 @@
+/* Compartment from which we call the switcher to perform inter-compartment
+ * transition. The call is via a capability, to update the PCC bounds
+ * appropriately to cover `switch_compartment`.
+ */
+
+// Compartment functions
+.global comp_f_fn
+.global comp_g_fn
+
+// Labels for size computations
+.global comp_f_fn_end
+.global comp_g_fn_end
+
+.type comp_f_fn, "function"
+comp_f_fn:
+    // Set compartment ID we want to switch to
+    mov       x0, #1
+
+    // Try to set the CLR to jump to a malicious function
+    adr       x10, comp_f_malicious
+    cvtp      c30, x10
```

Comment:
Right, so this is the crux of the demonstration: the result (in `c30`/`clr`) is out of bounds. It would (hopefully) fail if you simply returned immediately with `ret clr`.

What this example shows, at least to me, is that the compartment switching mechanism doesn't break that guarantee.

Is my understanding correct?

Reply:
That is correct. But to elaborate further, what this example is attempting to show is that the compartment, in some way, affects the environment that `switch_compartment` functions in. In this instance, it can affect the return address of the switcher. This is a sketch of how such an interaction could be subverted.

Further, this example can give rise to the following questions:
* what would be required for this approach to succeed in improperly modifying the return address;
* are there any other ways of deriving a `clr` such that the securities that work in this example are subverted/